### DISCUSSION TITLE
v7.3.4 Release of PowerShell

### DISCUSSION THREAD

**Discussion by daxian-dbw** [PowerShell/PowerShell#19503] [Show and tell]
### Engine Updates and Fixes

- Add instrumentation to `AmsiUtil` and make the `init` variable readonly (#18727)
- Fix support for `NanoServer` due to the lack of AMSI (#18882)
- Adding missing guard for telemetry optout to avoid `NullReferenceException` when importing modules (#18949) (Thanks @powercode!)
- Fix `VtSubstring` helper method to correctly check chars copied (#19240)
- Fix `ConciseView` to handle custom `ParserError` error records (#19239)

### Build and Packaging Improvements

<details>

<summary>

<p>Bump to use .NET 7.0.5</p>

</summary>

<ul>
<li>Update ThirdPartyNotices.txt</li>
<li>Update cgmanifest.json</li>
<li>Fix the template that creates nuget package</li>
<li>Update the wix file</li>
<li>Update to .NET SDK 7.0.203</li>
<li>Skip VT100 tests on Windows Server 2012R2 as console does not support it (#19413)</li>
<li>Improve package management acceptance tests by not going to the gallery (#19412)</li>
<li>Fix stage dependencies and typo in release build (#19353)</li>
<li>Fix issues in release build and release pipeline (#19338)</li>
<li>Restructure the package build to simplify signing and packaging stages (#19321)</li>
<li>Test fixes for stabilizing tests (#19068)</li>
<li>Add stage for symbols job in Release build (#18937)</li>
<li>Use reference assemblies generated by dotnet (#19302)</li>
<li>Add URL for all distributions (#19159)</li>
</ul>

</details>

[7.3.4]: https://github.com/PowerShell/PowerShell/compare/v7.3.3...v7.3.4

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - A3449062A5122D1BBF9BC8A06948F08FF6AA47729EBDAE807D46B270003C0DBD
- powershell_7.3.4-1.deb_amd64.deb
  - 1447A6D90AA66616757290E8A4BDE7DF0EE14D9B77ABA8D73021E849AD12F418
- powershell-7.3.4-1.cm.x86_64.rpm
  - FEB688B2D0E10B884D98BFA1433F96D63C54D62EC1AEF01849D25D0464B369B3
- powershell-7.3.4-1.rh.x86_64.rpm
  - 5B9902D320BFA13E75289539C1B73C8B827D195CDFDEADB71477EE5FBD374305
- powershell-7.3.4-linux-alpine-x64.tar.gz
  - 63F1402633B1FB2BDD89CCC89ADEDAC485365227FEFAC3120FE88BED1BBF53EE
- powershell-7.3.4-linux-arm32.tar.gz
  - C5E5B4F157283BFE30E44FD7468FAB26AF86926D58BA36BC406ED71C5C59F639
- powershell-7.3.4-linux-arm64.tar.gz
  - 34EE04D93394218C941499A2DF3289CCEC8FD3FAD33D366025971E6D536449F9
- powershell-7.3.4-linux-x64-fxdependent.tar.gz
  - 6B7016111455562291C548647B5DA7F4601522D3D7DBB5D0774888CC7A6C2B25
- powershell-7.3.4-linux-x64.tar.gz
  - E85D5544E13A924F8B2C4A5DC2D43ABE46E46633F89E8D138D39C0AAEACB9976
- powershell-7.3.4-osx-arm64.pkg
  - AFD80F3AEBD68AD2B00977DCEBC4C08D1E651D0763449935CE50AA3BF655E40E
- powershell-7.3.4-osx-arm64.tar.gz
  - 0E0EC5445E629D19F33FAB630E12201D32752761105E77A0AA29993CAE7BE164
- powershell-7.3.4-osx-x64.pkg
  - 1A09511F1CEC1CC491354413EAFF3502641F0CAF4E5A6F317B038830B9F1E53F
- powershell-7.3.4-osx-x64.tar.gz
  - 157DCEC95CF053E32CDAD823A59E1D3DD8DE271D87E79E357506408E12379F55
- PowerShell-7.3.4-win-arm32.zip
  - 2C5C6AE9E1C739C50AAC633C316ED9D006C7FFDA83ED8D8D85CFF5A69477C2FF
- PowerShell-7.3.4-win-arm64.zip
  - 08A4825AE39CFEB3EE9D35A5643918AA655EE349F43CFCE509FF7D5927752F34
- PowerShell-7.3.4-win-fxdependent.zip
  - 23636602351F9E12DBBE8CA252883933408192EC1ECC4DA512916BC81A9885BE
- PowerShell-7.3.4-win-fxdependentWinDesktop.zip
  - 790617195DE2B89F22F096372CEED4A4EF8C132EC94840D1C2EA88B73EB4905C
- PowerShell-7.3.4-win-x64.msi
  - 61B31EC847D4FDC4D39050F4F650968C55ACF6F16CB66B1F00BF05DB4E946559
- PowerShell-7.3.4-win-x64.zip
  - DBBA20DE5EC7388E4D70BFBD4359463D996DF0B7521BA38F7598C2A7A3115C6F
- PowerShell-7.3.4-win-x86.msi
  - F408178D37AE91D80ED58362FF729B97C1839008DCD1383AB97190316D9F8661
- PowerShell-7.3.4-win-x86.zip
  - 7933B2D50F86C3C252474B5AD3BD4CC3E94AAEA4CDA1600D6C787C92561B4796

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.4'>v7.3.4 Release of PowerShell</a>.</em>

**Comment by ogzhntutucu** [comment]
How do I upgrade the program to version 7.3.4?

In my terminal it says:
 A new PowerShell stable release is available: v7.3.4
 Upgrade now, or check out the release page at:
 https://aka.ms/PowerShell-Release?tag=v7.3.4

**Reply by diecodev** [reply]
```powershell
winget upgrade Microsoft.Powershell

# or

winget install --id Microsoft.Powershell --source winget
```
#### The commands above are not working (at least on my computer). Also, there isn't a Microsoft Store update available either.

**Reply by YoussefBenlarbi** [reply]
same issue as @diecodev

**Reply by diecodev** [reply]
@YoussefBenlarbi @thwisse If you have installed PowerShell from the Microsoft Store, you will need to uninstall PowerShell v7 and then open PowerShell v5. After that, use the following command: ```winget install --id Microsoft.PowerShell --source winget```. This way, you can easily upgrade your PowerShell version in future updates with the script ```winget upgrade Microsoft.PowerShell```

---

**Note:** If you have your own custom $profile file to customize your PowerShell experience, don't worry about it. If you follow the commands above, the file will not be deleted)

**Reply by ogzhntutucu** [reply]
In my terminal it says:
A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.

**Reply by ogzhntutucu** [reply]
when I type "winget upgrade" in my terminal:
Name                            Id                                      Version            Available     Source
PowerShell                    Microsoft.PowerShell        7.3.3.0             7.3.4.0         winget

**Comment by iota-008** [comment]
how to upgrade this without installing a new one but, upgrade the current one

**Reply by doctordns** [reply]
You have options (what follows is on Windows).  Personally, I really prefer using the `Install-PowerShell.ps1` script., This script does what it says: and allows you to install the latest released version, the latest preview versions, or the build of the day./ The later two are probably for the more hardcore - personally, I usually load the latest preview and upgrade the daily build version 1-2 times a week.  This script gives you all the flexibility and control you need to do things manually. 

Also, when you install via an MSI, you can get the option to have Windows Update able to service PowerShell. It can take a week or two AFTER GitHub has an updated version (and the metadata that says so) before Windows Update can update PowerShell. It would be nice for this to be a lot faster, but it isn't.

On Windows, you can either install PowerShell 7 to a "well-known place" (i.e. one of the folders within the $env:path variable). And the MSI can also extend the path, for example when you install PowerShell the very first time. When a new version comes along (eg you have 7.2.3 and 7.2.4 is released), then running the MSI just overwrites the older version with the newer one.

If you need to install specific versions of PowerShell side by side, the `Install-Powershell.ps1` can help. For much older versions, you may need to DIY it via the products release page on GitHub (https://github.com/PowerShell/PowerShell/releases). You can download and expand the .ZIP file into a version-specific folder (on my system, I have c:\PSDailyBuild and c:\PSPreview folders.

**Reply by iota-008** [reply]
Thanks!

**Comment by JerryABCD06** [comment]
How about the Microsoft Store version? There's still no update.

**Reply by diecodev** [reply]
You are right, there is no update available in the Microsoft Store yet. If you want to update your PowerShell version to the latest one, I recommend following the steps in my message above: https://github.com/orgs/PowerShell/discussions/19503#discussioncomment-5625187

**Reply by JerryABCD06** [reply]
Thank you very much. Could you tell me why?
Will the MS Store version get updates anymore?

**Reply by diecodev** [reply]
I cannot answer those questions becouse I am not a Microsoft employee

**Reply by doctordns** [reply]
The MSStore, and MU both work,but it takes a few days.

**Comment by TheOriginalPiff** [comment]
I’ll update bi-weekly.Sent from my iPhoneOn Apr 15, 2023, at 23:36, JerryABCD06 ***@***.***> wrote:﻿
Thank you very much. Could you tell me why?
Will the MS Store version get updates anymore?

**Reply by JerryABCD06** [reply]
OK. Thank you.

**Comment by TheOriginalPiff** [comment]
TOP secret, if you must know.. we can conduct an interview in person.Sent from my iPhoneOn Apr 16, 2023, at 00:51, Sachchang Phal ***@***.***> wrote:﻿I’ll update bi-weekly.Sent from my iPhoneOn Apr 15, 2023, at 23:36, JerryABCD06 ***@***.***> wrote:﻿
Thank you very much. Could you tell me why?
Will the MS Store version get updates anymore?

**Comment by Abcks05** [comment]
Bonjour je me rend compte que mon navigateur sur Android smart phone et tracé tout le temps que dois je faire

**Comment by armando-herastang** [comment]
PowerShell successfully updated automatically to v.7.3.4 on windows on an installation from the Microsoft Store.

**Comment by Manta45** [comment]
The Installation of PowerShell-7.3.4-win-x64.msi gives error message:
"Der Systemadministrator hat Richtlinien erlassen, um diese Installation zu verhindern."
Actually is PowerShel 7.3.3 installed and running. What is the reason? And how to prevent this error?

![Screenshot 2023-04-20 113235](https://user-images.githubusercontent.com/66775281/233324262-3c235558-ef16-4aaf-971b-9b6c32ea02c4.png)

**Comment by Jess-Lu-G** [comment]
So to download the latest version must we download every link or can there just be perhaps a zip folder with everything in please? Thanks! :)

**Reply by doctordns** [reply]
If you use the install-powershell.ps1 script it just works. Downloads the correct MSI and executes it. Job done!
